processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java

```diff
@@ -609,6 +613,9 @@ public static Dictionary getDictionary(AbsoluteTableIdentifier absoluteTableIden
           blockAssignmentStrategy = BlockAssignmentStrategy.BLOCK_SIZE_FIRST;
         } else {
           blockAssignmentStrategy = BlockAssignmentStrategy.BLOCK_NUM_FIRST;
+          // fall back to BLOCK_NUM_FIRST strategy need to reset
+          // the average expected size for each node
+          sizePerNode = numberOfBlocksPerNode;
```

Comment:
instead of reassigning the same variable, assignBlocksByDataLocality () can use numberOfBlocksPerNode directly?

Reply:
assignLeftOverBlocks also needs this similar if else self checks. I think its ok, you can   take a call to refactor now or later.